ui/org.eclipse.pde.ui/src/org/eclipse/pde/internal/ui/parts/PluginVersionPart.java

```diff
@@ -454,4 +486,43 @@ public void addListeners(ModifyListener minListener, ModifyListener maxListener)
 	protected String getGroupText() {
 		return PDEUIMessages.DependencyPropertiesDialog_groupText;
 	}
+
+	private enum Type {
+        MAJOR, MINOR, MICRO
+    }
+
+    private void bumpVersion(Type type) {
+        String target;
+        try {
+            Version v = new Version(getMinVersion());
+            int maj = v.getMajor(), min = v.getMinor(), mic = v.getMicro();
+            switch (type) {
+                case MAJOR:
+                    maj++;
+                    min = 0;
+                    mic = 0;
+                    break;
+                case MINOR:
+                    min++;
+                    mic = 0;
+                    break;
+                case MICRO:
+                    mic++;
+                    break;
+            }
+
+			String originalText = getMinVersion();
+			String qualifier = ""; //$NON-NLS-1$
+			int qualifierIndex = originalText.indexOf(v.toString()) + v.toString().length();
+			if (qualifierIndex < originalText.length()) {
+				qualifier = originalText.substring(qualifierIndex);
+			}
+
+			target = new Version(maj, min, mic).toString() + qualifier;
+        } catch (IllegalArgumentException e) {
+            target = "1.0.0"; //$NON-NLS-1$
```

Comment:
Instead of handling exception, maybe the input should be validated in advance and for example the buttons should be disabled when the field is empty/no version is specified yet?